Sources/FluentKit/Properties/OptionalParent.swift

```diff
@@ -48,17 +48,23 @@ extension OptionalParent: AnyProperty {
         var container = encoder.singleValueContainer()
         if let parent = self.eagerLoadedValue {
             try container.encode(parent)
-        } else {
+        } else if let id = self.id {
             try container.encode([
-                To.key(for: \._$id): self.id
+                To.key(for: \._$id): id
             ])
+        } else {
+            try container.encode(self.id)
         }
     }
 
     func decode(from decoder: Decoder) throws {
-        let container = try decoder.container(keyedBy: _ModelCodingKey.self)
-        try self.$id.decode(from: container.superDecoder(forKey: .string(To.key(for: \._$id))))
-        // TODO: allow for nested decoding
+        do {
+            let container = try decoder.container(keyedBy: _ModelCodingKey.self)
+            try self.$id.decode(from: container.superDecoder(forKey: .string(To.key(for: \._$id))))
+            // TODO: allow for nested decoding
+        } catch {
```

Comment:
There are more possible error cases being thrown away here than just the ID missing. One example is mismatched type. We should either catch the missing key error specifically or, better yet, use the methods on the decoder and container to check if the key is present.